source/fab/application.py

```diff
@@ -56,3 +57,32 @@ def run(self, source: Path):
             for filename in files:
                 print('    found in: ' + str(filename))
                 print('    depends on: ' + str(fortran_db.depends_on(unit)))
+
+
+class Dump(object):
+    def __init__(self, workspace: Path):
+        self._workspace = workspace
+        self._state = SqliteStateDatabase(workspace)
+
+    def run(self, stream=sys.stdout):
+        file_view = FileInfoDatabase(self._state)
+        print("File View", file=stream)
+        for filename in file_view.get_all_filenames():
+            file_info = file_view.get_file_info(filename)
+            print(f"  File   : {file_info.filename}", file=stream)
+            # Where files are generated in the working directory
+            # by third party tools, we cannot guarantee the hashes
+            if file_info.filename.match(f'{self._workspace}/*'):
```

Comment:
Noting the duplication here with what the main application is doing (albeit in a temporary way) - could we not have the main application produce its output via the `Dump` class?  I can see arguments for both - maybe not tying the two together like that is for the best (although again, it's only temporary)

Reply:
I think it best to keep them separate. The dumper behaviour is the desired behaviour for that tool. It absolutely isn't the desired behaviour for the build tool. I don't think we gain anything by conflating them other than drawing a false equivalence.

Reply:
Yeah I agree - just thought I'd mention it to show I was paying attention!